File: service-reservation/models/Reservation.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/db");

const Reservation = sequelize.define(
  "Reservation",
  {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },
    client_id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: "client", // Nom de la table Client
        key: "id",
      },
    },
    chambres: {
      type: DataTypes.JSON, // Liste d'IDs des chambres
      allowNull: false,
    },
    date_arrivee: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    date_depart: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    statut: {
      type: DataTypes.STRING,
      defaultValue: "En attente",
    },
    montant_total: {
      type: DataTypes.FLOAT,
      allowNull: true,
    },
    hebergement_id: {
      type: DataTypes.BIGINT, // Au lieu de INTEGER
      references: {
        model: "hebergement", // Nom de la table référencée
        key: "id",
      },
    },
  },
  {
    tableName: "reservation",
    timestamps: false,
  }
);

module.exports = Reservation;

File: DemandeProp (1)/DemandeProp/src/main/java/com/example/demandeprop/Data/Reservation.java
package com.example.demandeprop.Data;

import com.vladmihalcea.hibernate.type.json.JsonBinaryType;
import com.vladmihalcea.hibernate.type.json.JsonType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.hibernate.annotations.Type;


import java.util.List;

@Entity
@Table(name = "reservation")
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private Integer clientId;
    private String dateArrivee;
    private String dateDepart;
    private String statut;
    private Double montantTotal;
    private Long hebergementId;
    @Type(JsonType.class) // Utilise le type JSON pour le mapping
    @Column(columnDefinition = "json", nullable = false)
    private List<Long> chambres; // Liste des IDs des chambres
}
